src/poly/dcrt/sampler/trapdoor.rs

```diff
@@ -197,44 +212,137 @@ impl PolyTrapdoorSampler for DCRTPolyTrapdoorSampler {
 
         debug_mem("SetMatrixElement target_matrix_ptr completed");
 
-        let preimage_matrix_ptr = DCRTSquareMatTrapdoorGaussSamp(
+        let preimage_block_id_path = format!("data/{}.bin", preimage_block_id);
+
+        DCRTSquareMatTrapdoorGaussSampToFs(
             n as u32,
             k as u32,
             &public_matrix_ptr,
             &trapdoor.ptr_trapdoor,
             &target_matrix_ptr,
             2_i64,
             SIGMA,
+            &preimage_block_id_path,
         );
 
         debug_mem("DCRTSquareMatTrapdoorGaussSamp completed");
 
-        let nrow = size * (k + 2);
-        let ncol = size;
+        let mut path = PathBuf::new();
+        path.push(preimage_block_id_path);
+        path
 
-        let mut matrix_inner = Vec::with_capacity(nrow);
-        for i in 0..nrow {
-            let mut row = Vec::with_capacity(ncol);
-            for j in 0..ncol {
-                let poly = GetMatrixElement(&preimage_matrix_ptr, i, j);
-                let dcrt_poly = DCRTPoly::new(poly);
-                row.push(dcrt_poly);
-            }
-            matrix_inner.push(row);
+        // let nrow = size * (k + 2);
+        // let ncol = size;
+
+        // let mut matrix_inner = Vec::with_capacity(nrow);
+        // for i in 0..nrow {
+        //     let mut row = Vec::with_capacity(ncol);
+        //     for j in 0..ncol {
+        //         let poly = GetMatrixElement(&preimage_matrix_ptr, i, j);
+        //         let dcrt_poly = DCRTPoly::new(poly);
+        //         row.push(dcrt_poly);
+        //     }
+        //     matrix_inner.push(row);
+        // }
+
+        // debug_mem("GetMatrixElement completed");
+
+        // let full_preimage = DCRTPolyMatrix::from_poly_vec(params, matrix_inner);
+
+        // debug_mem("full_preimage generated");
+
+        // if target_cols < size {
+        //     debug_mem("Slicing full_preimage columns");
+        //     full_preimage.slice_columns(0, target_cols)
+        // } else {
+        //     full_preimage
+        // }
+    }
+
+    fn preimage_from_fs(
+        &self,
```

Comment:
actually we don't need to pass self on here , so that eval also don't need to pass Trapdoor struct to just use preimage from fs

Reply:
@enricobottazzi Also, could you fix this point as it is better to remove the arguments to the eval function?

Reply:
Actually, none of the methods on `DCRTPolyTrapdoorSampler` make any meaningful use of `Self`. The role of the struct, if any, is just to define the generic types defined in the `PolyTrapdoorSampler` trait and keep such related functions together. So I propose to maintain it as it is.

Reply:
@enricobottazzi In the future PR, I think we can define functions to load/store a matrix from/into a file with a compressed format in c++ as `load_compressed_file` and `store_compressed_file` functions in the `PolyMatrix` trait, as they are agnostic to the sampler.
Also, these function can be used to store the other vectors and matrices in the obfuscation struct.